src/GovUk.Education.ExploreEducationStatistics.Content.Services/MethodologyService.cs

```diff
@@ -125,5 +131,82 @@ private async Task<List<MethodologyVersionSummaryViewModel>> BuildMethodologiesF
                 await _methodologyVersionRepository.GetLatestPublishedVersionByPublication(publicationId);
             return _mapper.Map<List<MethodologyVersionSummaryViewModel>>(latestPublishedMethodologies);
         }
+
+        // This method is responsible for keeping Methodology Titles and Slugs in sync with their owning Publications
+        // where appropriate.  Methodologies always keep track of their owning Publication's titles and slugs for
+        // optimisation purposes.
+        public async Task PublicationTitleOrSlugChanged(Guid publicationId, string originalSlug, string updatedTitle,
+            string updatedSlug)
+        {
+            var slugChanged = originalSlug != updatedSlug;
+
+            var ownedMethodology = await _contentDbContext
+                .PublicationMethodologies
+                .Include(pm => pm.Methodology.LatestPublishedVersion)
+                .Include(pm => pm.Methodology.Versions)
+                .Where(pm => pm.PublicationId == publicationId && pm.Owner)
+                .Select(pm => pm.Methodology)
+                .SingleOrDefaultAsync();
+
+            if (ownedMethodology == null)
```

Comment:
I can't tell if this is actually a bad request or not at this point?  Do you think this is a valid method call if the supplied parametes produce no owned Methodology, or should we opt for either an ArgumentException or the returning of an Either<ActionResult, Unit> and do a `.SingleOrDefaultAsync().OrNotFound()`?

Reply:
I think it is correct. `PublicationTitleOrSlugChanged` is called when the publication's title or slug is changed. It's valid that the publication may not own a methodology.